enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1031 @@
+  Also using this API one can specify the header name in the list only once to avoid providing two entries of the same header name.
+
+#### For headers such as `X-Forwarded-For`, it makes sense to add an "Append" action. So why does this enhancement not cover appending?
+  Answer: Currently, we are proceeding with Set and Delete only. The RFE requires these two only. We have confirmed this with the stakeholders as well.
+  For the `X-Forwarded-*` headers, we have a dedicated API field for IngressControllers and a Route annotation, which provide the equivalent of an `Append` action for these headers:
+
+```sh
+$ oc explain ingresscontroller.spec.httpHeaders.forwardedHeaderPolicy
+KIND:     IngressController
+VERSION:  operator.openshift.io/v1
+
+FIELD:    forwardedHeaderPolicy <string>
+
+    DESCRIPTION:
+    forwardedHeaderPolicy specifies when and how the IngressController sets the
+    Forwarded, X-Forwarded-For, X-Forwarded-Host, X-Forwarded-Port,
+    X-Forwarded-Proto, and X-Forwarded-Proto-Version HTTP headers. The value
+    may be one of the following:
+    * "Append", which specifies that the IngressController appends the headers,
+    preserving existing headers.
+    * "Replace", which specifies that the IngressController sets the headers,
+    replacing any existing Forwarded or X-Forwarded-* headers.
+    * "IfNone", which specifies that the IngressController sets the headers if
+    they are not already set.
+    * "Never", which specifies that the IngressController never sets the
+    headers, preserving any existing headers. By default, the policy is
+    "Append".
+```
+
+#### Are some headers blocked from being set or deleted?
+  Answer: Please refer to the `Risks and Mitigations` section for more details.
+
+#### Many applications serve mixed content: static assets (images, js, css) and dynamic content (html, json). They usually want to cache the static assets on the client and do not care about session stickiness for those requests. Only the dynamic requests need to stick to the same server (session sticky)
+Most applications will set something like this for static assets:
+```sh
+cache-control: max-age=7200, must-revalidate, public
+```
+But the OpenShift router (HAProxy) will add cache-control: private which results in both headers in the response:
+```sh
+cache-control: private
+cache-control: max-age=7200, must-revalidate, public
+```
+This leads to issues with caching: CDNs like Cloudflare will not cache your files.
+Can we please make sure to only add the `cache-control: private` header if there is no `cache-control: public` header already present in the response?
+  Answer: A common technique is to put static assets on a different virtual host. In OpenShift terms, that means using a distinct route for static assets. This route can
+  be annotated with `haproxy.router.openshift.io/disable_cookies=true` to prevent setting the cookie and thus prevent setting cache-control.
```

Comment:
I think this is what you want to say here, but not sure.  It doesn't really explain why the answer is "no".

```suggestion
#### Can we please make sure to only add the `cache-control: private` header if there is no `cache-control: public` header already present in the response?

  Answer:  No.  Many applications serve mixed content: static assets (images, js, css) and dynamic content (html, json). They usually want to cache the static assets on the client and do not care about session stickiness for those requests. Only the dynamic requests need to stick to the same server (session sticky).

Most applications will set something like this for static assets:
```sh
cache-control: max-age=7200, must-revalidate, public
```
But the OpenShift router (HAProxy) will add cache-control: private which results in both headers in the response:
```sh
cache-control: private
cache-control: max-age=7200, must-revalidate, public
```
This can lead to issues with caching: e.g. CDNs like Cloudflare will not cache your files.

  As a workaround, a common technique is to put static assets on a different virtual host. In OpenShift terms, that means using a distinct route for static assets. This route can
  be annotated with `haproxy.router.openshift.io/disable_cookies=true` to prevent setting the cookie and thus prevent setting cache-control.
```

Reply:
done